testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/metadata/MavenMergeMetadataIT.java

```diff
@@ -0,0 +1,214 @@
+    }
+    if ((mr3 = repositoryManager.get("mr3")) == null) {
+      mr3 = repositoryManager.create(hostedConfig("mr3", VersionPolicy.RELEASE));
+    }
+    if ((mr4 = repositoryManager.get("mr4")) == null) {
+      mr4 = repositoryManager.create(hostedConfig("mr4", VersionPolicy.RELEASE));
+    }
+    if (repositoryManager.get("mavenGroup") != null) {
+      repositoryManager.delete("mavenGroup");
+    }
+  }
+
+  private void deployMetadataWithVersionTo(String version, Repository repository) throws Exception {
+    Metadata metadata = get(repository);
+    if (metadata == null) {
+      metadata = new Metadata();
+      metadata.setModelVersion("1.1.0");
+      metadata.setGroupId(GROUP_ID);
+      metadata.setArtifactId(ARTIFACT_ID);
+      metadata.setVersioning(new Versioning());
+    }
+    metadata.getVersioning().addVersion(version);
+    StringWriter sw = new StringWriter();
+    writer.write(sw, metadata);
+    final Maven2Client client = createAdminMaven2Client(repository.getName());
+    HttpResponse response = client.put(METADATA_PATH, new StringEntity(sw.toString(), ContentType.TEXT_XML));
+    assertThat(response.getStatusLine().getStatusCode(), equalTo(201));
+  }
+
+  @Nullable
+  private Metadata get(final Repository repository) throws Exception {
+    final Maven2Client client = createAdminMaven2Client(repository.getName());
+    HttpResponse response = client.get(METADATA_PATH);
+    if (response.getStatusLine().getStatusCode() == 200) {
+      Metadata metadata = reader.read(response.getEntity().getContent());
+      EntityUtils.consume(response.getEntity());
+      return metadata;
+    }
+    return null;
+  }
+
+  private Repository updateGroup(Repository group, Repository... members) throws Exception {
+    Configuration conf = group.getConfiguration();
+    final Set<String> memberNames = Sets
+        .newHashSet(Iterables.transform(Arrays.asList(members), new Function<Repository, String>()
+        {
+          @Override
+          public String apply(final Repository input) {
+            return input.getName();
+          }
+        }));
+    conf.attributes("group").set("memberNames", memberNames);
+    return repositoryManager.update(conf);
+  }
+
+  @Test
+  public void mergeMetadataOnMemberChange() throws Exception {
+    deployMetadataWithVersionTo("1.0", mr1);
+    deployMetadataWithVersionTo("2.0", mr2);
+    deployMetadataWithVersionTo("3.0", mr3);
+    deployMetadataWithVersionTo("4.0", mr4);
+    mavenGroup = repositoryManager.create(
+        groupConfig(
+            "mavenGroup",
+            mr1.getName(),
+            mr2.getName(),
+            mr3.getName(),
+            mr4.getName()
+        )
+    );
+    Metadata metadata = get(mavenGroup);
+    assertThat(metadata, notNullValue());
+    assertThat(metadata.getVersioning(), notNullValue());
+    assertThat(metadata.getVersioning().getVersions(), hasItems("1.0", "2.0", "3.0", "4.0"));
```

Comment:
@kellyrob99 Thanks, this change revealed other bug in the IT (not in the test subject but IT code itself), fixing that too. Related changes will make it clearer now regarding prepare method too